R/delete_project.R

```diff
@@ -34,47 +33,86 @@ delete_project <- function(project_id, conn) {
         log_event_table
      from redcap_projects
      where project_id in (%s)",
-     paste0(project_id, collapse = ",")
+      paste0(project_id, collapse = ",")
     )
   )
 
-  # select projects for deletion
   projects_to_delete <- redcap_projects[is.na(redcap_projects$date_deleted), ]
   redcap_project_ids <- projects_to_delete$project_id
   redcap_log_tables <- projects_to_delete$log_event_table
 
+
   if (nrow(projects_to_delete) > 0) {
-    tryCatch({
-      deleted_projects <- DBI::dbExecute(
-        conn,
-        sprintf(
-          "update redcap_projects set date_deleted = now() where project_id in (%s)",
-          paste0(redcap_project_ids, collapse = ",")
-        )
-      )
-    }, error = function(error_message) {
-      print(error_message)
-      return(FALSE)
-    })
+    delete_sql <- sprintf(
+      "UPDATE redcap_projects SET date_deleted = NOW() WHERE project_id IN (%s)",
+      paste0(redcap_project_ids, collapse = ",")
+    )
 
-    # log the event
-    tryCatch({
-      inserted_rows <- purrr::map2(
-        redcap_log_tables,
-        redcap_project_ids,
-        ~ DBI::dbExecute(
-          conn,
-          sprintf(
-            "insert into %s (object_type, event, project_id, description)
-             values ('redcap_projects', 'MANAGE', %d, 'delete project')",
-        .x,
-        .y)
+    tryCatch(
+      {
+        deleted_projects <- DBI::dbExecute(conn, delete_sql)
+      },
+      error = function(error_message) {
+        print(error_message)
+        return(FALSE)
+      }
+    )
+
+    # Define logging parameters
+    ts <- format(Sys.time(), "%Y%m%d%H%M%S") # Time stamp
+    user <- "admin" # Placeholder for user
```

Comment:
Test this with an invalid username to see how REDCap's UI will behave.

Then, revise the code to default it first to, the value in sys.getenv("API_USER"), then--if invalid usernames are allowed--the script name.